gem/migrations/0014_convert_recomended_articles.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def create_recomended_articles(main_article, article_list):
+    '''
+    Creates recommended article objects from article_list
+    and _prepends_ to existing recommended articles.
+    '''
+    from molo.core.models import ArticlePageRecommendedSections
+
+    existing_recommended_articles = [
+        ra.recommended_article.specific
+        for ra in main_article.recommended_articles.all()]
+    ArticlePageRecommendedSections.objects.filter(page=main_article).delete()
+
+    for hyperlinked_article in article_list:
+        ArticlePageRecommendedSections(
+            page=main_article,
+            recommended_article=hyperlinked_article).save()
+
+    # re-create existing recommended articles
+    for article in existing_recommended_articles:
+        if article not in article_list:
+            ArticlePageRecommendedSections(
+                page=main_article,
+                recommended_article=article).save()
+
+
+def convert_articles(apps, schema_editor):
+    '''
+    Derived from https://github.com/wagtail/wagtail/issues/2110
+    '''
+    from molo.core.models import ArticlePage
+    from wagtail.wagtailcore.blocks import StreamValue
+
+    articles = ArticlePage.objects.all()
+
+    for article in articles:
+        stream_data = []
+        linked_articles = []
+        for block in article.body.stream_data:
+            if block['type'] == 'page':
```

Comment:
Have you thought about the following article layout and the implications of this migration:
```
Paragraph 1
Paragraph 2
Paragraph 3
Link to Page
Paragraph 4
Paragraph 5
Link to Page
Link to Page
```
<img width="405" alt="screen shot 2017-04-08 at 1 13 25 pm" src="https://cloud.githubusercontent.com/assets/623417/24828423/2fcb101a-1c5d-11e7-9b5e-4bcc92cee1f8.png">
Also, removing the page link from the middle of an article will leave it in a bit of an un-readable state.

@saeedmb8 @nathanbegbie Please confirm that this is the expected behaviour with the content team.